homeassistant/components/plugwise/climate.py

```diff
@@ -48,94 +53,143 @@ def __init__(
     ) -> None:
         """Set up the Plugwise API."""
         super().__init__(coordinator, device_id)
-        self._attr_extra_state_attributes = {}
         self._attr_unique_id = f"{device_id}-climate"
+        self.hc_data = self.devices[self.gateway["heater_id"]]
 
-        # Determine preset modes
-        self._attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
         if presets := self.device.get("preset_modes"):
-            self._attr_supported_features |= ClimateEntityFeature.PRESET_MODE
             self._attr_preset_modes = presets
 
-        # Determine hvac modes and current hvac mode
-        self._attr_hvac_modes = [HVACMode.HEAT]
-        if self.coordinator.data.gateway.get("cooling_present"):
-            self._attr_hvac_modes.append(HVACMode.COOL)
-        if self.device.get("available_schedules") != ["None"]:
-            self._attr_hvac_modes.append(HVACMode.AUTO)
-
-        self._attr_min_temp = self.device.get("lower_bound", DEFAULT_MIN_TEMP)
-        self._attr_max_temp = self.device.get("upper_bound", DEFAULT_MAX_TEMP)
-        if resolution := self.device.get("resolution"):
+        self._attr_min_temp = self.device["thermostat"].get(
+            "lower_bound", DEFAULT_MIN_TEMP
+        )
+        self._attr_max_temp = self.device["thermostat"].get(
+            "upper_bound", DEFAULT_MAX_TEMP
+        )
+        if resolution := self.device["thermostat"].get("resolution", 0.1):
             # Ensure we don't drop below 0.1
             self._attr_target_temperature_step = max(resolution, 0.1)
 
     @property
-    def current_temperature(self) -> float | None:
+    def current_temperature(self) -> float:
         """Return the current temperature."""
-        return self.device["sensors"].get("temperature")
+        return self.device["sensors"]["temperature"]
 
     @property
-    def target_temperature(self) -> float | None:
-        """Return the temperature we try to reach."""
-        return self.device["sensors"].get("setpoint")
+    def hvac_action(self) -> HVACAction:
```

Comment:
Re-ordering of methods makes this PR unneeded and hard to review, it messes with the diffs. Let's not do that in this PR. Refactoring isn't needed for the dependency bump and should be done separately.

Same applies to all other methods moved.